flask/app/participants.py

```diff
@@ -247,6 +248,65 @@ def list_participants(page: int, limit: int) -> Response:
     abort(406, "Only 'text/csv' and 'application/json' HTTP accept headers supported")
 
 
+@participants_blueprint.route("/api/participants/<int:id>", methods=["GET"])
+@login_required
+def get_participant(id: int):
+
+    user = get_current_user()
+
+    # make sure the participant exists and if not send a 404
+    models.Participant.query.filter(
+        models.Participant.participant_id == id
+    ).first_or_404()
+
+    query = models.Participant.query.filter(
+        models.Participant.participant_id == id
+    ).options(
+        selectinload(models.Participant.family),
+        selectinload(models.Participant.institution),
+    )
+
+    if not user.is_admin:
+        query = filter_datasets_by_user_groups(
+            query.join(models.Participant.tissue_samples)
+            .join(models.TissueSample.datasets)
+            .options(
+                contains_eager(models.Participant.tissue_samples).contains_eager(
+                    models.TissueSample.datasets
+                ),
+            ),
+            user,
+        )
+
+    try:
+        participant = query.one()
+    except NoResultFound:
+        abort(403)
```

Comment:
If you look at `get_dataset` (`GET /api/datasets/:id`), you shouldn't be performing two queries here against the participant table. You should only do one. The only return code should be a 404, regardless of whether the user has permission or not to view a participant, as we don't want to reveal its existence by a different status code.

The family and institution relationships should be loaded by a joinedload, since a participant only has one of each.

Reply:
I made the corresponding changes! I am still using `query.one()` instead of `query.first_or_404()` because `first()` applies a LIMIT of 1 *within* the generated SQL, so that only one primary entity row is generated on the server side. This behaviour is not desired since we want to obtain all the datasets from the subquery.

Reply:
The main entity is the participant, but there is a subquery performed in `filter_datasets_by_user_groups`. This [post](https://stackoverflow.com/questions/29161730/what-is-the-difference-between-one-and-first) is relevant in terms of the difference between `first()` and `one()` in this context. 

Also, the [official documentation](https://docs.sqlalchemy.org/en/14/orm/query.html#sqlalchemy.orm.Query.first) says:

" first() applies a limit of one within the generated SQL, so that only one primary entity row is generated on the server side (note this may consist of multiple result rows if join-loaded collections are present)."

This is not the case for `one()`. I am not totally sure regarding my current implementation. (I also think this scenario is different than `get_dataset()`).